### ISSUE TITLE
[Bug]: Addon Styling resets all theme state in 7.0.0

### ISSUE THREAD

**Issue by gabalicious** [storybookjs/storybook#21865]
### Describe the bug

I upgraded to 7.0.0 for storybook and storybook addons.  We have two toolbar dropdowns for theme. One for light/dark theme (Tailwind css) and then one for client theme.  When we use the client theme and/or light/dark theme dropdowns there are some inconsistencies with how global state changes.

When in docs, Selecting Client Theme dropdown sets the light/dark theme as well for some reason.  Also when in docs, when we select the light/dark dropdown the client theme gets reset.

When in different variant pages of the component it seems most of the time my client theme doesn't change.

I guess the expected behavior in my opinion is that when selecting client theme it should not change the state of light/dark theme and vice versa.

### To Reproduce

Install the following repo with`npm I` and then `npm run dev` with node 18+

https://github.com/gabalicious/ui-workspace-example

Then toggle between client themes and dark mode themes while looking at the button component 

### System

```shell
System:
    OS: macOS 13.3
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 18.14.2 - ~/.asdf/installs/nodejs/18.14.2/bin/node
    Yarn: 1.22.19 - ~/.asdf/shims/yarn
    npm: 9.6.2 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 111.0.5563.146
    Firefox: 111.0.1
    Safari: 16.4
```

**Comment by gabalicious** [comment]
@Integrayshaun, So I still think there is an issue but with the variant part of the story.

Instructions to duplicate the issue
1. Select Primary Button under documentation
2. Try to change Client theme
3. It will not change.


Expected Behavior
1. Toggling between client1, client2, client3, and default should change the button.

**Comment by Integrayshaun** [comment]
@gabalicious right, I'm just closing this issue as your current problem is unrelated to `@storybook/addon-styling`. We'll continue to help you out on Discord 😁
